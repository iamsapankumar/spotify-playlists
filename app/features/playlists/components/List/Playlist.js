import React from 'react';
import { FlatList } from 'react-native';
import PropTypes from 'prop-types';
import { noop } from '@babel/types';
import PlaylistItem from '../ListItem/Playlistitem';

const getItemLayout = (data, index) => ({
  length: 30,
  offset: 44 * index,
  index,
});
const Playlist = ({
  onPress, playlistData = [],
}) => (
  <>
    <FlatList
      removeClippedSubviews
      data={playlistData}
      maxToRenderPerBatch={10}
      updateCellsBatchingPeriod={20}
      initialNumToRender={5}
      keyExtractor={(item, index) => index.toString()}
      windowSize={10}
      getItemLayout={getItemLayout}
      renderItem={(el) => (
        <PlaylistItem
          keyExtractor={(item) => item.key}
          playlistName={el.playlistName}
          onPress={onPress}
        />
      )}
    />

  </>
);

Playlist.propTypes = {
  onPress: PropTypes.func,
  playlistData: PropTypes.array,
};
Playlist.defaultProps = {
  onPress: noop,
  playlistData: [],
};

export default Playlist;
