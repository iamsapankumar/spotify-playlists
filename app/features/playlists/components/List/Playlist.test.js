/* eslint-disable no-undef */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Playlist from './Playlist';

it('renders correctly', () => {
  renderer.create(<Playlist />);
});
describe('Playlist', () => {
  describe('Rendering', () => {
    it('should match to snapshot =>', () => {
      const component = shallow(<Playlist />);
      expect(component).toMatchSnapshot();
    });
  });
});
