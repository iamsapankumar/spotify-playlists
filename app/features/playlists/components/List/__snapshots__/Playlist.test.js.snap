// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist Rendering should match to snapshot => 1`] = `ShallowWrapper {}`;
